mobile/src/main/java/ch/epfl/sdp/mobile/ui/play/ExpandableFloatingActionButton.kt

```diff
@@ -0,0 +1,240 @@
+package ch.epfl.sdp.mobile.ui.play
+
+import androidx.compose.animation.AnimatedContent
+import androidx.compose.animation.ExperimentalAnimationApi
+import androidx.compose.animation.animateColorAsState
+import androidx.compose.animation.core.animateFloatAsState
+import androidx.compose.animation.with
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.CornerSize
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.*
+import androidx.compose.runtime.*
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.util.lerp
+import ch.epfl.sdp.mobile.ui.FadeEnterTransition
+import ch.epfl.sdp.mobile.ui.FadeExitTransition
+import ch.epfl.sdp.mobile.ui.PawniesColors.Beige050
+import ch.epfl.sdp.mobile.ui.PawniesColors.Green100
+import ch.epfl.sdp.mobile.ui.PawniesColors.Green800
+
+/** The state of an [ExpandableFloatingActionButton] */
+interface ExpandableFloatingActionButtonState {
+
+  /** True iff the button should be currently expanded. */
+  var expanded: Boolean
```

Comment:
Technically, the `ExpandedFloatinActionButton` composable only has a single property (aka whether it's expanded) for now. Rather than exposing this, It's better to offer this state as a dedicated type (similarly to what we're doing with our screen view models for instance).

You can think of `ExpandableFloatingActionButtonState` as a type alias for `MutableState`, where the `MutableState.value` property is now called `expanded`. That's why we won't expose the `State` if we're actually already in an interface which represents state.